### DISCUSSION TITLE
August 2022 feedback

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4435] [Feedback]
I've been mostly copying existing code from one Canvas to another, trying to edit it.

- anything using fieldAccess is pretty unusable without traces
- when pasting function calls, the links on the right don't appear immediately
- it's really annoying that incomplete stuff is highlighted in red. Things seem more broken than they are. Especially true when names (eg record field or in `let`s) are highlighted in red.
- when pasting function calls, had a bunch of extra holes appear
- would have been useful to be able to copy entire handlers, especially including their traces
- some of our Slack usage has been failing for months/years, but we never knew. Notifications on error would be useful; I think also the slack functions don't error in a good way on failure
- I pressed enter at the end of the line, expecting a new line. Instead I got a pipe
- I pressed enter at the start of the line, expecting a new line. But nothing happened
- the overall editing experience feels janky and slow, especially compared to editing in VS code

**Comment by StachuDotNet** [comment]
> anything using fieldAccess is pretty unusable without traces

> would have been useful to be able to copy entire handlers, especially including their traces

An outline of goals towards this might be:
- clone handler
- with traces
- to another canvas

**Reply by pbiggar** [reply]
I was thinking that the copy buffer (ie copy and paste) might be enough for this.
